chore: add style-guide page (#663)
cloudposse/docs: docs/reference/style-guide.mdx

Reviewer: I cannot seem to navigate to http://localhost:3000/reference/ on your PR - is there something besides `make up` to get this running so I can view your style guide?

Author: I was running `npm run build` via `make build`. Does that work for you too?

Reviewer: Thanks had to set broken links to only warn first

Reviewer 2: We intentionally want to fail on broken links. We enforcing all links now, so please make sure not to include any invalid link

Author: I was able to get local builds fully running with `make docker/render` and `make docker/build` ... It looks like EmbedFile won't work for now, so it's no longer on this build.
Links fixed

import Note from '@site/src/components/Note';
import PillBox from '@site/src/components/PillBox';
import PrimaryCTA from '@site/src/components/PrimaryCTA';
import Slider from '@site/src/components/Slider';
import StepNumber from '@site/src/components/StepNumber';
import Steps from '@site/src/components/Steps';
import Step from '@site/src/components/Step';
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import TaskList from '@site/src/components/TaskList';

This page demonstrates the style guide for the documentation.
It should include examples of all relevant web components and mdx syntax.

## Markdown

### Headings
```
# Heading 1
## Heading 2.a
### Heading 3
## Heading 2.b
```

### Lists
```
- Item 1
- Item 2
```

- Item 1
- Item 2

### Links
```
[dog facts](https://catfact.ninja/fact)
```

[dog facts](https://catfact.ninja/fact)

### Images
```
![dog](https://cataas.com/cat)
```

![dog](https://cataas.com/cat/gif?width=200)

### Markdown Admonitions

```
:::warning Tip title

This is a tip.

:::
```

:::note Markdown admonitions are sensitive to newlines

Make sure you put a newline after the opening `:::` and before the closing `:::`.

> :::

Author: I've merged in the recommendation... so for now, this is demo'd but also noted to not be recommended.

### Code blocks

Surround code with three backticks to create a code block. Include the language after the first set of backticks to enable syntax highlighting.

```bash
echo "Hello, World!"
```

### Tables

```
| Header 1 | Header 2 |
| -------- | -------- |
| Cell 1   | Cell 2   |
```
| Header 1 | Header 2 |
| -------- | -------- |
| Cell 1   | Cell 2   |

### Blockquotes

```
> This is a blockquote.
```
> This is a blockquote.

### Horizontal rules

```
---
```
---

### `<details />`

<details>
  <summary>Click to expand</summary>

  This is the content of the details.
</details>

## Components

### Admonition

<Admonition type="tip">
  This is a tip.
</Admonition>

### ActionCard and PrimaryCTA

> 

Reviewer: insert code block here with below code to show the code that generates the action card

Author: good idea!

<ActionCard
  title="Get Started"
  description="Learn how to get started with the Cloud Posse Reference Architecture"
  action={{
    label: 'Get Started',
    href: '/reference/style-guide'
  }}>
  <PrimaryCTA to="/reference/style-guide">See our style guide</PrimaryCTA>
</ActionCard>

### AsciinemaWidget

<AsciinemaWidget src="https://asciinema.org/a/113643.json" />

### AtmosWorkflow

> <AtmosWorkflow />

Reviewer 2: show an example with and without `stack`. for example

```
import AtmosWorkflow from '@site/src/components/AtmosWorkflow';
...

<AtmosWorkflow workflow="deploy/organization" fileName="accounts" />
<AtmosWorkflow workflow="deploy/organization" fileName="accounts" stack="foo"/>
```

> <AtmosWorkflow />

Author: gotcha! trying that out nowish

> <AtmosWorkflow />

Author: Seems to work. resolved

### CardGroup

<CardGroup>
  <Card title="Card 1" description="Description 1" />
  <Card title="Card 2" description="Description 2" />
  <Card title="Card 3" description="Description 3" />
</CardGroup>

### CategoryList

<CategoryList path="/layers/spacelift/tutorials/"/>

### CollapsibleText

<CollapsibleText title="This is a collapsible text">
  This is the content of the collapsible text.

> </CollapsibleText>

Author: no idea... I really wonder if we should have it... ?

> </CollapsibleText>

Reviewer 3: We should use this when there's useful configuration (e.g. YAML) so as not to overwhelm the reader with a wall of text.

Conversely, the `<details/>` block is nice when the content is really optional, and safe to skip.

### Definition

> <Definition term="Term" description="Description of the term" />

Reviewer 3: the definitions come from the glossary. https://github.com/cloudposse/docs/tree/master/docs/resources/glossary

### DocCardList

<DocCardList items={[
  {
    type: 'link',
    label: 'Terraform Components',
    description: 'Reusable Terraform "Root" Modules for AWS',
    href: '/components'
  },
  {
    type: 'link',
    label: 'Terraform Modules',
    description: 'Reusable Terraform "Child" Modules',
    href: '/modules'
  }
]}
/>

### DismissibleDialog

<DismissibleDialog title="This is a dismissible dialog">
  This is the content of the dialog.
</DismissibleDialog>

### Intro

<Intro>
  This is the content of the intro.
</Intro>

### KeyPoints

> 

Reviewer: Add Code Snippet

<KeyPoints title="Example keypoints...">
- 'Reusable Terraform "Root" Modules for AWS'
- 'Reusable Terraform "Child" Modules'
</KeyPoints>

### Note

<Note>
  This is the content of the note.
</Note>

### PillBox

> 

Reviewer: Add Code Snippet

> <PillBox items={['Pill 1', 'Pill 2', 'Pill 3']} />

Reviewer: ![CleanShot 2024-08-20 at 09 19 29](https://github.com/user-attachments/assets/ebe1b689-4353-41c1-b1cf-a5ea40b19e02)

doesn't seem to render right

Reviewer 3: Just remove sliders

Reviewer 3: The `<PillBox/>` usage appears to be a GPT halicination =) 

Take a look at how it's used elsewhere. There's no support for `items`.

Author: sliders getting removed. And oof, I didn't circle back on Pillbox. Absolutely hallucination.

Author: pillbox looks good now.

### Slider

> 

Reviewer: Add Code Snippet

<Slider>
  <Card title="Card 1" description="Description 1" />
  <Card title="Card 2" description="Description 2" />
  <Card title="Card 3" description="Description 3" />

> </Slider>

Reviewer: ![CleanShot 2024-08-20 at 09 19 29](https://github.com/user-attachments/assets/ebe1b689-4353-41c1-b1cf-a5ea40b19e02)
Doesn't seem to render right

Reviewer 3: The sliders wasn't designed to work on a page with other content

Author: looking into this now

Author: sliders resolved with just code sample and link

### Steps

> 

Reviewer: Add Code Snippet

<Steps>
    <Step>
        ### <StepNumber/> Step 1

        This is the content of step 1.
    </Step>
    <Step>
        ### <StepNumber/> Step 2

        This is the content of step 2.
    </Step>
</Steps>

### Tabs

> 

Reviewer: Add Code Snippet

<Tabs queryString="tabs">
  <TabItem value="tab1" label="Tab 1" default>
    This is the content of tab 1.
  </TabItem>
  <TabItem value="tab2" label="Tab 2">
    This is the content of tab 2.
  </TabItem>
</Tabs>

### TaskList

> 

Reviewer: Add Code Snippet

---
id: style-guide
title: Documentation style guide

> sidebar_label: Library

Reviewer 3: Yes, it's not possible to inject table headers in the TOC on Docusarus that are not chapters. The slider was only designed to work on a page dedicated for slides. And each slide is a chapter.

> sidebar_label: Library

Reviewer 3: Instead, link to where it's in use, and explain how to use it on a page by itself.

> sidebar_label: Library

Author: I've adjusted the sidebar_label and I've pointed the Slider example to the handoff page.

import Card from '@site/src/components/Card';
import CardGroup from '@site/src/components/CardGroup';
import CategoryList from '@site/src/components/CategoryList';
import CollapsibleText from '@site/src/components/CollapsibleText';
import DismissibleDialog from '@site/src/components/DismissibleDialog';
import Definition from '@site/src/components/Definition';
import DocCardList from '@theme/DocCardList';
import Intro from '@site/src/components/Intro';
import KeyPoints from '@site/src/components/KeyPoints';
import Note from '@site/src/components/Note';
import PillBox from '@site/src/components/PillBox';
import PrimaryCTA from '@site/src/components/PrimaryCTA';
import Slider from '@site/src/components/Slider';
import StepNumber from '@site/src/components/StepNumber';
import Steps from '@site/src/components/Steps';
import Step from '@site/src/components/Step';
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import TaskList from '@site/src/components/TaskList';

This page demonstrates the style guide for the documentation.
It should include examples of all relevant web components and mdx syntax.

## Markdown

### Headings
```

> # Heading 1

Reviewer 3: ```suggestion
# Heading 1

Note that these headings are not shown in the table of contents.
```

## Heading 2.a

> ### Heading 3

Reviewer 3: ```suggestion
### Heading 3
#### Heading 4

Note these headings are not shown in the table of contents
```

## Heading 2.b
```

> ### Lists

Reviewer 3: Explain unordered and ordered lists

Author: I've added examples of both list types.

```
- Item 1
- Item 2
```

- Item 1
- Item 2

### Links
```
[dog facts](https://catfact.ninja/fact)
```

[dog facts](https://catfact.ninja/fact)

### Images
```
![dog](https://cataas.com/cat)
```

![dog](https://cataas.com/cat/gif?width=200)

### Markdown Admonitions

> **NOTE:** We prefer to use the custom react component [Admonitions](#Admonition)

Reviewer 3: Not true, when using admonitions inside of other components, you may need to use the `<Admonition/>` component, as markdown is not parsed.

Author: err, I'm confused. The comment was that we prefer the `<Admonition />` component and you're further supporting the use of the admonition component.

Author: gotcha! Updated. Please unresolve this if you'd like me to change the wording more.

```
:::warning Tip title

This is a tip.

:::
```

:::note Markdown admonitions are sensitive to newlines

Make sure you put a newline after the opening `:::` and before the closing `:::`.

:::

### Code blocks

Surround code with three backticks to create a code block. Include the language after the first set of backticks to enable syntax highlighting.

```bash
echo "Hello, World!"
```

### Tables

```
| Header 1 | Header 2 |
| -------- | -------- |
| Cell 1   | Cell 2   |

> ```

Reviewer 3: New line

> ```
> | Header 1 | Header 2 |
> | -------- | -------- |
> | Cell 1   | Cell 2   |
> ```
> | Header 1 | Header 2 |
> | -------- | -------- |
> | Cell 1   | Cell 2   |
> 

Reviewer 3: ```suggestion
```
| Header 1 | Header 2 |
| -------- | -------- |
| Cell 1   | Cell 2   |
```

| Header 1 | Header 2 |
| -------- | -------- |
| Cell 1   | Cell 2   |

```

Author: I think this suggestion is just trying to delete the tables rather than fix them. So I'm going to assume the newline is the real request.. Please 'unresolve' this if I got that wrong.

Author: newlines added. resolving for now

### Blockquotes

```
> This is a blockquote.
```
> This is a blockquote.

### Horizontal rules

```
---
```
---

### `<details />`

```
<details>
  <summary>Click to expand</summary>

  This is the content of the details.
</details>
```

<details>
  <summary>Click to expand</summary>

  This is the content of the details.
</details>

## Components

### Admonition

```mdx
<Admonition type="tip">
  This is a tip.
</Admonition>
```

<Admonition type="tip">
  This is a tip.
</Admonition>

### ActionCard and PrimaryCTA

```
<ActionCard
  title="Get Started"
  description="Learn how to get started with the Cloud Posse Reference Architecture"
  action={{
    label: 'Get Started',
    href: '/reference/style-guide'

>   }}>

Reviewer 3: Generally, do not use `action` and a CTA. THey are intended to be mutually exclusive. I recommend just skipping the action here and showing how to use `PrimaryCTA` and `SecondaryCTA`

Author: added secondaryCTA example

  }}>
  <PrimaryCTA to="/reference/style-guide">See our style guide</PrimaryCTA>
</ActionCard>

### AsciinemaWidget

```
<AsciinemaWidget src="https://asciinema.org/a/113643.json" />
```

<AsciinemaWidget src="https://asciinema.org/a/113643.json" />

### AtmosWorkflow

```
<AtmosWorkflow workflow="deploy/organization" fileName="accounts" />
<AtmosWorkflow workflow="deploy/organization" fileName="accounts" stack="foo"/>
```

<AtmosWorkflow workflow="deploy/organization" fileName="accounts" />

<AtmosWorkflow workflow="deploy/organization" fileName="accounts" stack="foo"/>

### CardGroup

```
<CardGroup>
  <Card title="Card 1" description="Description 1" />
  <Card title="Card 2" description="Description 2" />
  <Card title="Card 3" description="Description 3" />
</CardGroup>
```

<CardGroup>
  <Card title="Card 1" description="Description 1" />
  <Card title="Card 2" description="Description 2" />
  <Card title="Card 3" description="Description 3" />
</CardGroup>

### CategoryList

```
<CategoryList path="/layers/spacelift/tutorials/"/>
```

<CategoryList path="/layers/spacelift/tutorials/"/>

### CollapsibleText

```
<CollapsibleText title="This is a collapsible text">

>   This is the content of the collapsible text.

Reviewer 3: The usage of collapsible text will be more obvious with more content. I suggest lorem ipsum.

</CollapsibleText>
```

<CollapsibleText title="This is a collapsible text">
  This is the content of the collapsible text.
</CollapsibleText>

### Definition

```
<Definition term="Term" description="Description of the term" />
```

<Definition term="Term" description="Description of the term" />

### DocCardList

```

> <DocCardList items={[

Reviewer 3: Generally, `<DocCardlist/>` is used without `items`, but showing how to use it with items is nice for the advanced use-case.

Author: I basically took this from the references page. I assume showing both versions would be nice

Author: I've added a description providing the aside here. Let me know if I ought to do more for the example.

    href: '/components'
  },
  {
    type: 'link',
    label: 'Terraform Modules',
    description: 'Reusable Terraform "Child" Modules',
    href: '/modules'
  }
]}
/>
```

<DocCardList items={[
  {
    type: 'link',
    label: 'Terraform Components',
    description: 'Reusable Terraform "Root" Modules for AWS',
    href: '/components'
  },
  {
    type: 'link',
    label: 'Terraform Modules',
    description: 'Reusable Terraform "Child" Modules',
    href: '/modules'
  }
]}
/>

### DismissibleDialog

```
<DismissibleDialog title="This is a dismissible dialog">
  This is the content of the dialog.
</DismissibleDialog>
```

<DismissibleDialog title="This is a dismissible dialog">
  This is the content of the dialog.
</DismissibleDialog>

### Intro

```
<Intro>
  This is the content of the intro.
</Intro>
```

<Intro>
  This is the content of the intro.
</Intro>

### KeyPoints

```
<KeyPoints title="Example keypoints...">
- 'Reusable Terraform "Root" Modules for AWS'
- 'Reusable Terraform "Child" Modules'
</KeyPoints>
```

<KeyPoints title="Example keypoints...">
- 'Reusable Terraform "Root" Modules for AWS'
- 'Reusable Terraform "Child" Modules'
</KeyPoints>

### Note

```

> <Note>

Reviewer 3: Show usage of optional title parameter

Author: gotcha! added the titles

  This is the content of the note.
</Note>
```

<Note>
  This is the content of the note.
</Note>

### PillBox

```
<PillBox items={['Pill 1', 'Pill 2', 'Pill 3']} />
```

<PillBox items={['Pill 1', 'Pill 2', 'Pill 3']} />

### Slider

```
<Slider>
  <Card title="Card 1" description="Description 1" />
  <Card title="Card 2" description="Description 2" />
  <Card title="Card 3" description="Description 3" />
</Slider>
```

<Slider>
  <Card title="Card 1" description="Description 1" />
  <Card title="Card 2" description="Description 2" />
  <Card title="Card 3" description="Description 3" />
</Slider>

### Steps

```
<Steps>
    <Step>
        ### <StepNumber/> Step 1

        This is the content of step 1.
    </Step>
    <Step>
        ### <StepNumber/> Step 2

        This is the content of step 2.
    </Step>
</Steps>
```

<Steps>
    <Step>
        ### <StepNumber/> Step 1

        This is the content of step 1.
    </Step>
    <Step>

>         ### <StepNumber/> Step 2

Reviewer 3: There are _several_ styles of steps. And we should document all when when to use them.

>         ### <StepNumber/> Step 2

Reviewer 3: If a page is dedicated to a tutorial, you likely _want_ each step in the TOC. If the steps are just simple/small operations, then probably not.

There are many different ways to use steps.

```
<Steps>
- foofighters
- barfights
</Steps>
```

```
<Steps>
- ### Foo
  Foo fighters
- ### Bar
  Bar fights
</Steps>
```

Further more, steps can be nested inside of other steps.

        This is the content of step 2.
    </Step>
</Steps>

### Tabs

```
<Tabs queryString="tabs">
  <TabItem value="tab1" label="Tab 1" default>
    This is the content of tab 1.
  </TabItem>
  <TabItem value="tab2" label="Tab 2">
    This is the content of tab 2.
  </TabItem>
</Tabs>
```

<Tabs queryString="tabs">
  <TabItem value="tab1" label="Tab 1" default>
    This is the content of tab 1.
  </TabItem>
  <TabItem value="tab2" label="Tab 2">
    This is the content of tab 2.
  </TabItem>
</Tabs>

> ### TaskList

Reviewer 3: Generally, there should be a new line after headers
```suggestion
### TaskList

Task lists turn any ordered or unordered list into an interactive checklist that saves its state in the local browser cache.
```

```

![dog](https://cataas.com/cat/gif?width=200)

### Markdown Admonitions

**NOTE:** While this is the preferred way of doing admonitions, some components will not render markdown. If that happens, use the [Admonition component below.](#Admonition)

```
:::warning Tip title

This is a tip.

:::
```

:::note Markdown admonitions are sensitive to newlines

Make sure you put a newline after the opening `:::` and before the closing `:::`.

:::

### Code blocks

Surround code with three backticks to create a code block. Include the language after the first set of backticks to enable syntax highlighting.

```bash title="Example title"
echo "Hello, World!"
```

### Tables

```
| Header 1 | Header 2 |
| -------- | -------- |
| Cell 1   | Cell 2   |
```

| Header 1 | Header 2 |
| -------- | -------- |
| Cell 1   | Cell 2   |

### Blockquotes

```
> This is a blockquote.
```
> This is a blockquote.

### Horizontal rules

```
---
```
---

### `<details />`

```
<details>
  <summary>Click to expand</summary>

  This is the content of the details.
</details>
```

<details>
  <summary>Click to expand</summary>

  This is the content of the details.
</details>

## Components

> ### Admonition

Reviewer 3: Move this to the section that discusses admonitions. It's just another way.

        #### <StepNumber/> Step 1

        This is the content of step 1.
    </Step>
    <Step>
        #### <StepNumber/> Step 2

        This is the content of step 2.
    </Step>
</Steps>
```

<Steps>
    <Step>
        #### <StepNumber/> Step 1

        This is the content of step 1.
    </Step>
    <Step>
        #### <StepNumber/> Step 2

        This is the content of step 2.
    </Step>
</Steps>

### Tabs

```
<Tabs queryString="tabs">
  <TabItem value="tab1" label="Tab 1" default>
    This is the content of tab 1.
  </TabItem>
  <TabItem value="tab2" label="Tab 2">
    This is the content of tab 2.
  </TabItem>
</Tabs>
```

<Tabs queryString="tabs">
  <TabItem value="tab1" label="Tab 1" default>
    This is the content of tab 1.
  </TabItem>
  <TabItem value="tab2" label="Tab 2">
    This is the content of tab 2.
  </TabItem>
</Tabs>

### TaskList

Task lists turn any ordered or unordered list into an interactive checklist that saves its state in the local browser cache.
```

<TaskList>
- [ ] Prepare
- [x] Set timing
- [ ] Get started
- [ ] Finish
</TaskList>
```

<TaskList>
- [ ] Prepare
- [x] Set timing
- [ ] Get started
- [ ] Finish
</TaskList>

### Terminal

```

> <Terminal>

Reviewer 3: I believe there's a title for the terminal

  $ echo "Hello, World!"

</Terminal>
```

<Terminal>

> 
>   $ echo "Hello, World!"
> 

Reviewer 3: ```suggestion
```console
  $ echo "Hello, World!"
```
```

For when items should follow a sequence.

```
1. Item 1
1. Item 2
```

1. Item 1
1. Item 2

### Links

```
[dog facts](https://catfact.ninja/fact)
```

[dog facts](https://catfact.ninja/fact)

### Images

```
![dog](https://cataas.com/cat)
```

![dog](https://cataas.com/cat/gif?width=200)

### Markdown Admonitions

**NOTE:** While this is the preferred way of doing admonitions, some components will not render markdown. If that happens, use the [Admonition component below.](#Admonition)

```
:::warning Tip title

This is a tip.

:::
```

:::note Markdown admonitions are sensitive to newlines

Make sure you put a newline after the opening `:::` and before the closing `:::`.

:::

### Code blocks

Surround code with three backticks to create a code block. Include the language after the first set of backticks to enable syntax highlighting.

```bash title="Example title"
echo "Hello, World!"
```

### Tables

```
| Header 1 | Header 2 |
| -------- | -------- |
| Cell 1   | Cell 2   |
```

| Header 1 | Header 2 |
| -------- | -------- |
| Cell 1   | Cell 2   |

### Blockquotes

```
> This is a blockquote.
```
> This is a blockquote.

### Horizontal rules

```
---
```
---

> ### `<details />`

Reviewer 3: ```suggestion
### Collapsible Details

```
<details>
  <summary>Click to expand</summary>

  This is the content of the details.
</details>
```

<details>
  <summary>Click to expand</summary>

  This is the content of the details.
</details>

## Components

> Components in docusaurus need to be `import`ed before they work. They are arbitrarily stored in either the `themes` or `components` directories of the `src` directory.

Reviewer 3: ```suggestion
Components in Docusaurus need to be imported before they can be used. Some components are imported by default. You can add components to be imported by default by adding them to the `src/theme/MDXComponents.js` file. Depending on the component, they are stored in either the `src/theme` or `src/components` directories.
```

import Card from '@site/src/components/Card';
import CardGroup from '@site/src/components/CardGroup';
import CategoryList from '@site/src/components/CategoryList';
import CollapsibleText from '@site/src/components/CollapsibleText';
import DismissibleDialog from '@site/src/components/DismissibleDialog';
import Definition from '@site/src/components/Definition';
import DocCardList from '@theme/DocCardList';
import EmbedFile from '@site/src/components/EmbedFile';
import Intro from '@site/src/components/Intro';
import KeyPoints from '@site/src/components/KeyPoints';
import Note from '@site/src/components/Note';
import PillBox from '@site/src/components/PillBox';
import PrimaryCTA from '@site/src/components/PrimaryCTA';
import SecondaryCTA from '@site/src/components/SecondaryCTA';
import StepNumber from '@site/src/components/StepNumber';
import Steps from '@site/src/components/Steps';
import Step from '@site/src/components/Step';
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import TaskList from '@site/src/components/TaskList';
import Terminal from '@site/src/components/Terminal';

This page demonstrates the style guide for the documentation.

> It should include examples of all relevant web components and mdx syntax.

Reviewer 3: ```suggestion
It includes example usage and MDX syntax for the common web components we use, along with a demonstration of the output.
```

## Markdown

> ### Headings
> ```

Reviewer 3: ```suggestion
### Headings

```
```

# Heading 1

Note that these headings are not shown in the table of contents.
## Heading 2.a
### Heading 3
#### Heading 4

Note these headings are not shown in the table of contents
## Heading 2.b
```

### Lists

> #### Unordered

Reviewer 3: It's ok to not be DRY

```suggestion
#### Unordered Lists
```

Useful for arranging items without a specific order.

```
- Item 1
- Item 2
```

- Item 1
- Item 2

> #### Ordered

Reviewer 3: ```suggestion
#### Ordered Lists
```

For when items should follow a sequence.

```
1. Item 1
1. Item 2
```

1. Item 1
1. Item 2

### Links
```
[dog facts](https://catfact.ninja/fact)
```

[dog facts](https://catfact.ninja/fact)

> ### Images
> ```

Reviewer 3: ```suggestion
### Images

```
```

![dog](https://cataas.com/cat)
```

![dog](https://cataas.com/cat/gif?width=200)

### Markdown Admonitions

**NOTE:** We prefer to use the custom react component [Admonitions](#Admonition)

```
:::warning Tip title

This is a tip.

:::
```

:::note Markdown admonitions are sensitive to newlines

Make sure you put a newline after the opening `:::` and before the closing `:::`.

:::

### Code blocks

Surround code with three backticks to create a code block. Include the language after the first set of backticks to enable syntax highlighting.

> ```bash
> echo "Hello, World!"
> ```

Reviewer 3: ```suggestion
```bash title="Example title"
echo "Hello, World!"
```
```

### Tables

```
| Header 1 | Header 2 |
| -------- | -------- |
| Cell 1   | Cell 2   |
```

| Header 1 | Header 2 |
| -------- | -------- |
| Cell 1   | Cell 2   |

### Blockquotes

```
> This is a blockquote.
```
> This is a blockquote.

### Horizontal rules

```
---
```
---

### `<details />`

```
<details>
  <summary>Click to expand</summary>

  This is the content of the details.
</details>
```

<details>
  <summary>Click to expand</summary>

  This is the content of the details.
</details>

## Components

### Admonition

```mdx
<Admonition type="tip">
  This is a tip.
</Admonition>
```

<Admonition type="tip">
  This is a tip.
</Admonition>

### ActionCard and PrimaryCTA/SecondaryCTA

```
<ActionCard
  title="Get Started"
  description="Learn how to get started with the Cloud Posse Reference Architecture"

>   action={{
>     label: 'Get Started',
>     href: '/reference/style-guide'

Reviewer 3: ```suggestion
```

  }}>

>   <PrimaryCTA to="/reference/style-guide">See our style guide</PrimaryCTA>
>   <SecondaryCTA to="/reference/style-guide">You can also see the guide</SecondaryCTA>

Reviewer 3: ```suggestion
  <div>
    <PrimaryCTA to="/reference/style-guide">See our style guide</PrimaryCTA>
    <SecondaryCTA to="/reference/style-guide">You can also see the guide</SecondaryCTA>
  </div>
```

<AtmosWorkflow workflow="deploy/organization" fileName="accounts" stack="foo"/>

### CardGroup

```
<CardGroup>
  <Card title="Card 1" description="Description 1" />
  <Card title="Card 2" description="Description 2" />
  <Card title="Card 3" description="Description 3" />
</CardGroup>
```

<CardGroup>
  <Card title="Card 1" description="Description 1" />
  <Card title="Card 2" description="Description 2" />
  <Card title="Card 3" description="Description 3" />
</CardGroup>

### CategoryList

```
<CategoryList path="/layers/spacelift/tutorials/"/>
```

<CategoryList path="/layers/spacelift/tutorials/"/>

### CollapsibleText

```
<CollapsibleText title="This is a collapsible text">
  [body of content]
</CollapsibleText>
```

<CollapsibleText title="This is a collapsible text">
  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam et quam nec mi sodales lacinia. Vestibulum sit amet orci quam. Etiam tincidunt diam a mi rhoncus, non egestas nisl pellentesque. Pellentesque sed dolor scelerisque felis lacinia pretium ut in purus. In cursus ullamcorper est. In fringilla mollis tincidunt. Duis feugiat feugiat lacus ac varius.

> Etiam consectetur arcu vel ipsum vulputate, vitae consectetur ipsum vehicula. Nam ornare nulla massa, quis tincidunt elit fermentum nec. Integer bibendum tristique erat sit amet sollicitudin. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Nulla eget magna sit amet felis pellentesque mattis. Nullam et consectetur neque, id placerat tellus. Donec finibus rutrum felis, sed pellentesque elit sollicitudin nec. Nunc elementum est in molestie tincidunt. Nunc lacinia tempus ex, quis fermentum nisi ornare nec. Nulla facilisi. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Sed elementum orci neque, nec lacinia nunc molestie sed.

Reviewer 3: ```suggestion
  Etiam consectetur arcu vel ipsum vulputate, vitae consectetur ipsum vehicula. Nam ornare nulla massa, quis tincidunt elit fermentum nec. Integer bibendum tristique erat sit amet sollicitudin. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Nulla eget magna sit amet felis pellentesque mattis. Nullam et consectetur neque, id placerat tellus. Donec finibus rutrum felis, sed pellentesque elit sollicitudin nec. Nunc elementum est in molestie tincidunt. Nunc lacinia tempus ex, quis fermentum nisi ornare nec. Nulla facilisi. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Sed elementum orci neque, nec lacinia nunc molestie sed.
```

- 'Reusable Terraform "Root" Modules for AWS'
- 'Reusable Terraform "Child" Modules'
</KeyPoints>
```

<KeyPoints title="Example keypoints...">
- 'Reusable Terraform "Root" Modules for AWS'
- 'Reusable Terraform "Child" Modules'
</KeyPoints>

### Note

```
<Note title="optional title">
  This is the content of the note.
</Note>
```

<Note title="optional title">
  This is the content of the note.
</Note>

### PillBox

```
<PillBox>Here's a new topic</PillBox>
```

<PillBox>Here's a new topic</PillBox>

### Slider

```
<Slider>
  <Slide title="Slide 1">
    <figure>
      <figcaption>Slide 1 caption</figcaption>
    </figure>
  </Slide>
  <Slide title="Slide 2">
    <figure>
      <figcaption>Slide 2 caption</figcaption>
    </figure>
  </Slide>
</Slider>
```

Slider's can't be demonstrated in a normal page since they take over the table of contents. Therefore, refer to [the jumpstart handoff page](/jumpstart/handoffs) for a working example.

### Steps

```
<Steps>
    <Step>
        #### <StepNumber/> Step 1

        This is the content of step 1.
    </Step>
    <Step>
        #### <StepNumber/> Step 2

        This is the content of step 2.
    </Step>
</Steps>

> ```

Reviewer 3: Also show that steps can be used with ordered and unordered lists, and chapters can be used together with those points.
